### DISCUSSION TITLE
displaying dissaggregated data on the map

### DISCUSSION THREAD

**Discussion by christoph4711** [open-sdg/open-sdg#1405] [Ideas]
@brockfanning 
Is it possible to display the dissagregated data in the map?

I have data about housing in lower saxony [here](https://christoph4711.github.io/open-sdg-site-starter/4-4-1/)

On the map currently the floor space in squaremeters of the flats is displayed. In the data I have also the amount of 1-, 2-, and 3-room flats. Is it possible to display e.g. the amount of 1-room or 2-room flats in the map based on the selection in the table/graph view?

**Comment by brockfanning** [comment]
@christoph4711 This would be a great feature. It's tricky though, because the disaggregation drop-downs that we use allow more than one disaggregation to be selected, but maps would only be capable of displaying one at a time.

For example, a line graph can easily show "female" and "male" by having 2 lines. But the choropleth maps can only display "female" or "male", but not both.

So for maps, we need a special widget that allows only one selection (like a radio button). So even before we tackle the technical hurdles, there are some user-interface questions to figure out.

**Comment by christoph4711** [comment]
ok. I understand....

**Comment by christoph4711** [comment]
@brockfanning 
@MoBosse 
may be an interesting repository.....

https://github.com/floledermann/mapmap.js

https://github.com/floledermann/mapmap-examples

used [here](https://service.destatis.de/DE/karten/migration_integration_regionen.html): 

and this: https://github.com/d3/d3
used [here](https://wahlatlas.net/experimente/flaechen/)

**Comment by brockfanning** [comment]
After doing a revamp of the mapping system in preparation for Open SDG 1.0.0, a simple drop-down select is actually very easy to implement. But I think plenty of UX discussion is in order. Here is a screencast of what it would look like, with the code we already have. (For some reason in the video the drop-down doesn't expand, but you get the idea.)

![captured (8)](https://user-images.githubusercontent.com/1319083/81083641-d5ccf480-8ec2-11ea-9607-ab6cb89a69cf.gif)

@AnnCorp @phillipgardiner @SavvasStephanides @LucyGwilliamAdmin Just tagging for awareness.

**Comment by brockfanning** [comment]
Summary of notes from #769:
1. When there is a column for the region name which "mirrors" the GeoCode column, a random region appears in the disaggregation drop-down. A way to avoid this is to use "exclude_columns" in the data repository map_layer config, but this is not obvious. Can we automate that?
2. In IE, the disaggregation select stops working after the first choice.

**Comment by brockfanning** [comment]
One usability note about the prototype in #663: I happened to be testing on an iPhone, and found that because of the iPhone's treatment of drop-downs, it was pretty much unusable. I couldn't even really tell if it was working. Haven't tried Android yet.

**Comment by WohnungNDS** [comment]
with Android 7.1.1 it works....

![grafik](https://user-images.githubusercontent.com/67795376/87218022-73272780-c34f-11ea-985a-902eb5572682.png)

**Comment by WohnungNDS** [comment]
with iOS 13.5.1 on IPhone5 SE it works as well....

![grafik](https://user-images.githubusercontent.com/67795376/87218211-f1d09480-c350-11ea-80b0-0a4e7031816c.png)

**Comment by brockfanning** [comment]
Ah, thank you. I guess it's fine - I'm just not accustomed to the iPhone drop-down treatment I guess.

**Comment by WohnungNDS** [comment]
@brockfanning will this feature included in a future release? I am asking becuse I have spent some time in structuring my tables. If it will be included until October or so, then I leave my tables as they are.. else I have to think about to divide some of my tables to have a map for some disaggregated data...

**Comment by brockfanning** [comment]
@WohnungNDS I know we're just about to release 1.4.0, and it won't be in that. It could potentially go into 1.5.0, and I'll bring it up with the team. In the meantime, I've refreshed the pull-request (#663) with the latest code, in case it's possible for you to use that.

**Comment by WohnungNDS** [comment]
@brockfanning Great! Thank you! Do I have to point to `brockfanning:map-disaggregation-select` in my site [config file](https://github.com/WohnungNDS/IM_Site/blob/d3b145cbceab3a7373012c90a2ddcc12bf7af829/_config.yml#L31) and in my [requirements.txt](https://github.com/WohnungNDS/IM_Data/blob/91efd0265bf9f88e9efd2f2010db17938b01c95a/scripts/requirements.txt#L2) to use it?

**Comment by brockfanning** [comment]
@WohnungNDS To try it out you would need to set your `remote_theme` to `brockfanning/open-sdg@map-disaggregation-select` in your site configuration. Thank you for testing it - that PR hasn't been tested in a long time.

**Comment by WohnungNDS** [comment]
@brockfanning @LucyGwilliamAdmin 

min max Values

Something is wrong with the min / max values in the legend

![image](https://user-images.githubusercontent.com/67795376/140046059-8e14f82e-0dab-4ea4-8473-6ebb232195c0.png)

min = 10,1 and max = 10

[Indicator](https://wohnungnds.github.io/IM_Site/5-2-3/)
[Indicator data](https://github.com/WohnungNDS/IM_Data/blob/develop/data/indicator_5-3-2.csv)
[metadata](https://github.com/WohnungNDS/IM_Data/blob/develop/meta/5-3-2.md)

[site config](https://github.com/WohnungNDS/IM_Site/blob/develop/_config.yml)
[data config](https://github.com/WohnungNDS/IM_Data/blob/develop/config_data.yml)

also mentioned here #356 

may be wrong here... feel free to push it here #1391

**Comment by WohnungNDS** [comment]
A collegue has developed this [map ](https://www.nls.niedersachsen.de/Schulkarten/Schulen_in_Niedersachsen_2020.html)for different schools in lower saxony... may be an interesting option also for open-sdg maps
